transactions.go

```diff
@@ -145,6 +148,9 @@ type CureStatusTransaction struct {
 func (t CureStatusTransaction) Mutate(b *Battle) {
 	receiver := b.getPokemon(t.Target.party, t.Target.partySlot)
 	receiver.StatusEffects.clear(t.StatusEffect)
+	if t.StatusEffect.check(StatusSleep) {
+		delete(receiver.metadata, MetaSleepTime)
```

Comment:
(nit) I had this initialize and reset the meta field to 0 to avoid checking for existence in `b.preRound`.